tests/Feature/BookTest.php

```diff
@@ -266,4 +266,86 @@ public function test_book_claim_db_success()
             'claimed' => 1
         ]);
     }
-}
+
+    public function test_book_return_invalid_email()
+    {       
+        $response = $this->putJson('api/books/return/1', ['email' => 'test']);
+        $response->assertStatus(422);
+        $response->assertInvalid('email');
+    }
+
+    public function test_book_not_found_return()
+    {
+        $response = $this->putJson('/api/books/return/564756445323254', [
+            'email' => 'test@test.com'
+        ]);
+        $response->assertStatus(404);
+        $response->assertJson(function (AssertableJson $json) {
+            $json->has('message')
+                ->where(
+                    'message',
+                    'Book 564756445323254 was not found'
+                );
+        });
+    }
+
+    public function test_book_not_currently_claimed()
+    {
+        $book = Book::factory(['claimed_by_name' => null])->create();
+        
+        $response = $this->putJson("api/books/return/$book->id", [
+            'email' => 'test@test.com'
+        ]);
+
+        $response->assertStatus(400); 
+        $response->assertJson(function (AssertableJson $json) use($book) {
+            $json->has('message')
+                ->where(
+                    'message',
+                    "Book $book->id is not currently claimed"
+                );
+        });
+    }
+
+    public function test_book_incorrect__claimed_email()
+    {
+        $book = Book::factory()->create();
+
+        $response = $this->putJson("api/books/return/$book->id", [
+            'email' => 'wrong@wrong.com'
+        ]);
+        
+        $response->assertStatus(500); 
+        $response->assertJson(function (AssertableJson $json) use($book) {
+            $json->has('message')
+                ->where(
+                    'message',
+                    "Book $book->id was not able to be returned"
+                );
+        });
+    }
+
+    public function test_book_return_success()
+    {
+        $book = Book::factory()->create();
+
+        $response = $this->putJson("api/books/return/$book->id", [
+            'email' => $book->claimed_by_email
+        ]);
+
+        $response->assertOk();
+        $response->assertJson(function (AssertableJson $json) use($book) {
+            $json->has('message')
+                ->where(
+                    'message',
+                    "Book $book->id was returned"
+                );
+        });
+
+        $this->assertDatabaseHas('books', [
```

Comment:
You have two distinct tests in this function. Make a new test_book_return_db_success() and include 345-349 there... like we did with the test_book_claim_db_success()

Reply:
We were also unsure about whether this would be considered bad practice, so we ran it past Ash. He basically said either way is fine- this way gives slightly less info about which step (the actual PUT request or the response return) is causing the test to fail if the test fails, but it's definitely DRYer, so much of a muchness imo